FAQ.md

```diff
@@ -0,0 +1,112 @@
+    longer meets the constraints of your direct dependencies.
+
+    > If you absolutely need to specify the constraint of a transitive dep from your own project, you have two options:
+    >
+    > Specify the constraint on github.com/gorilla/context via an override. Overrides apply globally, but are a power only given to the root project, so if anything else imports your project, the override won't be used.
+    > Mark github.com/gorilla/context as a required package in the manifest. This will cause it to be treated as a direct dependency, and your constraint will come into effect.
+    >
+    > However, before taking either of those steps, I'd say it's worth asking if you actually need to use master of github.com/gorilla/context. I imagine it's imported by github.com/gorilla/mux - and if that package is OK with using the tagged release instead of master (which is the preferred mode of operation anyway), then maybe that should be good enough for you? If you really needed something out of github.com/gorilla/context, then you'd probably be importing it directly and doing something with it
+    -[@sdboyer in #385](https://github.com/golang/dep/issues/385#issuecomment-294361087)
+
+* Is package X included in the `ignored` list in your manifest?
+
+    Remove it from `ignored` and try again.
+
+## Why is `dep` ignoring the version specified in the manifest?
+Only direct dependencies can be managed with a `depenencies` entry
+in the manifest. Use an `overrides` entry for transitive dependencies.
+
+>  Constraints:
+>
+>  1. Can be declared by any project's manifest, yours or a dependency
+>  2. Apply only to direct dependencies of the project declaring the constraint
+>  3. Must not conflict with the constraints declared in any other project's manifest
+>
+>  Overrides:
+>
+>  1. Are only utilized from the current/your project's manifest
+>  2. Apply globally, to direct and transitive dependencies
+>  3. Supercede constraints declared in all manifests, yours or a dependency's
+>
+>  Overrides are also discussed with some visuals in [the gps docs](https://github.com/sdboyer/gps/wiki/gps-for-Implementors#overrides).
+-[@sdboyer in #328](https://github.com/golang/dep/issues/328#issuecomment-286631961)
+
+## `dep` deleted my files in the vendor directory!
+First, sorry! 😞 We hope you were able to recover your files...
+
+> dep assumes complete control of vendor/, and may indeed blow things away if it feels like it.
+-[@peterbourgon in #206](https://github.com/golang/dep/issues/206#issuecomment-277139419)
+
+## Can I put the manifest and lock in the vendor directory?
+No.
+
+> Placing these files inside vendor/ would concretely bind us to vendor/ in the long term.
+> We prefer to treat the use of vendor/ as an implementation detail.
+-[@sdboyer on go package management list](https://groups.google.com/d/msg/go-package-management/et1qFUjrkP4/LQFCHP4WBQAJ)
+
+## Unable to update checked out version: fatal: reference is not a tree
```

Comment:
Oooh yeah you're right. Here is the new behavior https://github.com/golang/dep/issues/405.